apps/dashboard/src/context/opt-in-provider.tsx

```diff
@@ -1,29 +1,21 @@
-import { PropsWithChildren, useLayoutEffect } from 'react';
+import { PropsWithChildren, useEffect } from 'react';
 import { NewDashboardOptInStatusEnum } from '@novu/shared';
 import { useNewDashboardOptIn } from '@/hooks/use-new-dashboard-opt-in';
-import { useSearchParams } from 'react-router-dom';
 
 export const OptInProvider = (props: PropsWithChildren) => {
   const { children } = props;
-  const { status, isLoaded, redirectToLegacyDashboard, optIn } = useNewDashboardOptIn();
-  const [searchParams] = useSearchParams();
-  const hasV2OptIn = searchParams.has('v2_opt_in');
-
-  useLayoutEffect(() => {
-    if (isLoaded && status !== NewDashboardOptInStatusEnum.OPTED_IN) {
-      if (hasV2OptIn) {
-        (async () => {
-          await optIn();
-        })();
-      } else {
-        redirectToLegacyDashboard();
-      }
-    }
+  const { status, isLoaded, redirectToLegacyDashboard } = useNewDashboardOptIn();
 
+  useEffect(() => {
     // set light theme on the new domain for both legacy and new dashboard
     localStorage.setItem('mantine-theme', 'light');
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [status, redirectToLegacyDashboard, isLoaded, hasV2OptIn]);
+  }, [status, redirectToLegacyDashboard, isLoaded]);
+
+  if (isLoaded && status !== NewDashboardOptInStatusEnum.OPTED_IN) {
+    redirectToLegacyDashboard();
+
+    return null;
+  }
```

Comment:
when the user is opted-out redirect him to the old dashboard with preserving the URL route and query params